clusterloader2/GETTING_STARTED.md

```diff
@@ -0,0 +1,239 @@
+
+```bash
+gvm linkthis k8s.io/perf-tests
+```
+
+## Create cluster using kind
+Follow kind installation guide [Kind install]
+
+Create cluster v1.21.1 with one master and one node:
+```
+kind create cluster --image=kindest/node:v1.21.1 --name=test-cluster --wait=5m
+```
+
+This command generates cluster access credentials, which are stored in `${HOME}/.kube/config` with context name test-cluster.
+
+## Prepare simple test to run
+Let's start with preparing test config (config.yaml):
+```yaml
+# Test name
+name: test
+# Number of namespaces that will be created
+namespace:
+  number: 1
+# Tuning set with qps 1 will execute 1 action every second.
+# All objects for given (namespace, replica numer) are treated as one unit of action.
+# You can check other tunning sets, their configuration and how they work here:
+# https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/tuningset/simple_tuning_set_factory.go#L52
+tuningSets:
+- name: Uniform1qps
+  qpsLoad:
+    qps: 1
+steps:
+# We want to start two measurement.
+# One will measure time it takes to start a pod, second will make sure we wait for pod to be in running state.
+# Setting the action field to "start" begins execution of measurement.
+- name: Start measurements
+  measurements:
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: start
+# This label specifies for which pods we want to measure pod startup latency.
+      labelSelector: group = test-pod
+# If 99 percentile of pod startup latency will go over threshold, this test will fail.
+      threshold: 120s
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+# Type of object we are waiting for
+      kind: Deployment
+      labelSelector: group = test-pod
+# How long we want to wait till object is in running state.
+      operationTimeout: 120s
+# Once we created these two measurements, we can create deployment in next step.
+- name: Create deployment
+  phases:
+# namespaceRange specifies in which namespaces we want to create this deployment.
+  - namespaceRange:
+      min: 1
+      max: 1
+# Here we an specify how many replicas of object we want to create within namespace.
+    replicasPerNamespace: 1
+    tuningSet: Uniform1qps
+    objectBundle:
+    - basename: test-deployment
+# Path to file containing template for object to be created.
+      objectTemplatePath: "deployment.yaml"
+# Parameters to fill template.
+      templateFillMap:
+        Replicas: 1
+        Group: test-pod
+        CpuRequest: 1m
+        MemoryRequest: 10M
+# Once we started measurements and created deployment, now we can wait for this deployment to be running.
+- name: Wait for pods to be running
+  measurements:
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: gather
+# Now we can delete deployment.
+- name: Delete deployment
```

Comment:
I know we generally want to do that, but I would really like the initial example to be as simple as possible.

Technically - deleting the deployment doesn't provide much additional value, so I would just remove it.